experimenter/experimenter/nimbus_ui_new/static/js/edit_branches.js

```diff
@@ -0,0 +1,50 @@
+import { basicSetup } from "codemirror";
+import { EditorView } from "@codemirror/view";
+import { json, jsonParseLinter } from "@codemirror/lang-json";
+import { linter } from "@codemirror/lint";
+import { autocompletion } from "@codemirror/autocomplete";
+import { schemaAutocomplete, schemaLinter } from "./validator.js";
+import $ from "jquery";
+
+const setupCodemirror = () => {
+  const textareas = document.querySelectorAll(".value-editor");
+
+  textareas.forEach((textarea) => {
+    const jsonSchema = JSON.parse(textarea.dataset["schema"]);
+
+    const extensions = [
+      basicSetup,
+      EditorView.updateListener.of((v) => {
+        if (v.docChanged) {
+          const value = v.state.doc.toString();
+          const textarea = v.view.dom.parentNode.querySelector(".value-editor");
+          textarea.value = value;
+        }
+      }),
```

Comment:
I guess we need to mirror this to a hidden input because the htmx stuff would otherwise blow away the contents of the codemirror view yeah?

Reply:
It's not cuz of the HTMX stuff, it's cuz Django emits a textarea from the form, but codemirror just creates its own element.  In the React UI you just use a hook to suck the value out of it and stick it in the obj you pass into the GQL calls, whereas Django is doing naive HTML Form POSTs, so this is the equivalent of pulling the value out of Codemirror into some obj for API calls, but writing back to the naive DOM element.